src/textual/css/stylesheet.py

```diff
@@ -177,6 +179,11 @@ def _parse_rules(self, css: str, path: str | PurePath) -> list[RuleSet]:
             raise
         except Exception as error:
             raise StylesheetError(f"failed to parse css; {error}")
+
+        if path in self._widget_css_paths:
```

Comment:
What if `_parse_rules` could take a boolean indicating that it was Widget CSS, and pass that in to `parse` which would store it in the rule? I think that would negate this check and loop.